src/Aperture/overlap.jl

```diff
@@ -154,3 +154,117 @@ function area_arc(x0, y0, x1, y1, r)
     θ = 2asin(0.5a / r)
     return 0.5r^2 * (θ - sin(θ))
 end
+
+"""
+Utility function to find whether a point is inside ellipse or not.
+
+If point inside ellipse: Returns true else returns false
+
+point_completely_inside_ellipse(x, y, h, k, cxx, cyy, cxy)
+
+x: x coordinate of the test point
+y: y coordinate of the test point
+h: x coordinate of the center of ellipse
+k: y coordinate of the center of ellipse
+cxx, cyy, cxy: coefficients of equation of ellipse
+
+General equation of ellipse:
+    cxx * (x - h)^2 + cxy * (x - h) * (y - k) + cyy * (y - k)^2 = 1
+"""
+function point_completely_inside_ellipse(x, y, h, k, cxx, cyy, cxy)
+    if cxx * (x - h)^2 + cxy *(x - h)*(y - k) + cyy * (y - k)^2  - 1 > 0
+        return false
+    else
+        return true
+    end
+end
```

Comment:
In you break this down a little bit, you've effectively written
```
if true; return true
if false; return false.
```

Therefore, I would just write
```julia
point_completely_inside_ellipse(x, y, h, k, cxx, cyy, cxy) = cxx * (x - h)^2 + cxy *(x - h)*(y - k) + cyy * (y - k)^2  - 1 > 0
```

Reply:
for a shorter function name, consider `intersects_ellipse`

Reply:
How about inside_ellipse? this function basically tells whether the point is inside ellipse or not.

Reply:
That's the same thing as `intersects`, geometrically. nothing wrong with either way.